src/evaluators/CallExpression.js

```diff
@@ -79,6 +100,21 @@ function evaluateReference(
     if (base.kind === "conditional") {
       return evaluateConditionalReferenceBase(ref, ast, strictCode, env, realm);
     }
+    let referencedName = ref.referencedName;
+
+    // When dealing with a PrimitiveValue, like StringValue, NumberValue, IntegralValue etc
+    // if we are referencing a prototype method, then it's safe to access, even
+    // on an abstract value as the value is immutable and can't have a property
+    // that matches the prototype method (unless the prototype was modified).
+    // In pure scope, we assume the global prototype of built-ins has not been altered.
+    let possiblePrimitiveProto = getPrimitiveProtoFromAbstractValueType(realm, base);
+    if (possiblePrimitiveProto !== undefined && typeof referencedName === "string") {
+      let possibleProtoDesc = possiblePrimitiveProto.$GetOwnProperty(referencedName);
```

Comment:
Do we also make the assumption in pure mode that we don't have properties on the abstract value with the same name as global prototype properties?

Reply:
They are primitive values, they can't have properties (that I'm aware of) on `string`, `number`, `boolean` and `symbol`. You can `[]` access characters at an index, but then that wouldn't be on the prototype.